### ISSUE TITLE
no-unused-vars error message should provide hint on ignorePattern.

### ISSUE THREAD

**Issue by TheSavior** [eslint/eslint#9171]
We use no-unused-vars but provide allowed patterns for ignored vars and args. 

```js
"no-unused-vars": [1, {"vars": "all", varsIgnorePattern: "^_", "args": "all", "argsIgnorePattern": "^_"}],
```

With this code:
```
/* eslint no-unused-vars: [2, {"vars": "all", varsIgnorePattern: "^_", "args": "all", "argsIgnorePattern": "^_"}] */

[1,2,3].map((item, index, array) => {
  return item / array.length;
});
```

We get this error:

```
3:20 - 'index' is defined but never used. (no-unused-vars)
```

It is unreasonable for our engineers to be aware of our allowed pattern in the configuration and this warning will instead cause people to go through lots of work and confusion to get eslint to pass, or they will just turn it off.

I think a more valuable warning message would be something like this:

```
3:20 - 'index' is defined but never used. If this is intentional, consider matching the allowed unused argument pattern /^_/ (no-unused-vars)
```

**Comment by platinumazure** [comment]
I'm :+1: to this as long as we can find a way to shorten the message a bit.

**Comment by TheSavior** [comment]
```
> 'is defined but never used. If this is intentional, consider matching the allowed unused argument pattern'.length
104

> 'is defined but never used. Allowed unused args must match'.length
57
```

How do you feel about that?

**Comment by platinumazure** [comment]
> How do you feel about that?

Works for me, thanks.

**Comment by TheSavior** [comment]
I modified the tests to reflect the new error messages. If those look good, we can update the rule to make them pass. I just want to make sure I'm covering the cases and people are happy with it before putting in the work on the rule.

**Comment by TheSavior** [comment]
What is the path forward to getting this tagged as "accepted" instead of "evaluating"?

**Comment by platinumazure** [comment]
@TheSavior We will accept the issue when enough people on the ESLint core team have supported the issue that we have reached [consensus](https://eslint.org/docs/maintainer-guide/issues.html#consensus). This means one champion and three :+1:s from other team members.

I'll champion. So we need two more :+1:s (mine no longer counts).

@eslint/eslint-team Anyone else want to support this? I think this would be a small but very useful improvement to the messaging for this rule.

**Comment by kaicataldo** [comment]
This is now accepted

**Comment by TheSavior** [comment]
Yay, great. Any more comments on the PR would be great, otherwise that is good to go as well. #9176

**Comment by kaicataldo** [comment]
It looks like we missed the `fixes` message on the PR. This was closed in https://github.com/eslint/eslint/pull/9176.
